partiql-eval/src/main/kotlin/org/partiql/eval/internal/helpers/DatumUtils.kt

```diff
@@ -0,0 +1,20 @@
+package org.partiql.eval.internal.helpers
+
+import org.partiql.spi.value.Datum
+import org.partiql.types.PType
+
+internal object DatumUtils {
+
+    /**
+     * Calls [Datum.lower] if the datum is a variant, otherwise returns the datum. If you don't know whether the value
+     * is of type [PType.VARIANT], you should use [Datum.lowerSafe] before invoking whatever methods you intend to use.
+     * This is essentially a workaround for the fact that we currently don't know whether a particular expression will be
+     * [PType.VARIANT] or not. This can eventually be optimized to accommodate this.
```

Comment:
nit: could we reword the last sentence to not refer to "this" and be more specific? Might be confusing as to what "this" is referring to